src/preprocess/run_tg.py

```diff
@@ -43,105 +66,143 @@
+            centralised_ids_filepath = superpop_split.get_ids_path(
                 fold_index=fold_index,
                 part_name='train',
-                node=node
+                node='ALL'  # centralised PCA
             )
 
-            ids.extend(pd.read_csv(ids_filepath, sep='\t')['IID'].to_list())
-
-        # Store the list of ids inside the super population split file structure
-        centralised_ids_filepath = superpop_split.get_ids_path(
-            fold_index=fold_index,
-            part_name='train',
-            node='ALL'  # centralised PCA
-        )
-
-        pd.DataFrame({'IID': ids}).to_csv(centralised_ids_filepath, sep='\t', index=False)
-
-    # 6. K-fold centralised PCA
-    for fold_index in range(FOLDS_NUMBER):
-        logger.info(f'Centralised PCA for fold {fold_index}')
-        run_plink(
-            args_list=[
-                '--pfile', os.path.join(SPLIT_GENO_DIR, 'ALL_filtered'),
-                '--keep', superpop_split.get_ids_path(fold_index=fold_index, part_name='train', node='ALL'),
-                '--freq', 'counts',
-                '--out', superpop_split.get_pca_path(node='ALL', fold_index=fold_index, part='train', ext=''),
-                '--pca', 'allele-wts', '20'
-            ]
-        )
-
-        logger.info(f'Projecting train, test, and val parts for each node for fold {fold_index}...')
-        for node in nodes:
-            for part_name in ['train', 'val', 'test']:
-                run_plink(
-                    args_list=[
-                        '--pfile', superpop_split.get_pfile_path(node=node, fold_index=fold_index, part_name=part_name),
-                        '--read-freq', superpop_split.get_pca_path(node='ALL', fold_index=fold_index, part='train', ext='.acount'),
-                        '--score', superpop_split.get_pca_path(node='ALL', fold_index=fold_index, part='train', ext='.eigenvec.allele'), '2', '5', 'header-read', 'no-mean-imputation', 'variance-standardize', '--score-col-nums', '6-25',
-                        '--out', superpop_split.get_pca_path(node=node, fold_index=fold_index, part=part_name),
-                        '--set-missing-var-ids', '@:#'
-                    ]
-                )
+            pd.DataFrame({'IID': ids}).to_csv(centralised_ids_filepath, sep='\t', index=False)
+
+    # 6. PCA
+    if Stage.FEDERATED_PCA in stages:
+        runner = FederatedPCASimulationRunner(
+            source_folder=SPLIT_GENO_DIR,
+            result_folder=FEDERATED_PCA_DIR,
+            variant_ids_file=os.path.join(SPLIT_GENO_DIR, 'ALL.prune.in'),
+            n_components=10,
+            method='P-STACK',
+            nodes=nodes
+        ).run()
+    elif Stage.CENTRALIZED_PCA in stages:
+        for fold_index in range(FOLDS_NUMBER):
+            logger.info(f'Centralised PCA for fold {fold_index}')
+            run_plink(
+                args_list=[
+                    '--pfile', TG_BFILE_PATH,
```

Comment:
Please change `TG_BFILE_PATH` which refers to raw unprocessed data and replaced it with `ALL_filtered`
```suggestion
                    '--pfile', os.path.join(SPLIT_GENO_DIR, 'ALL_filtered'),
```